sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileAsyncApiTests.java

```diff
@@ -603,6 +604,67 @@ public void uploadRangeFromURL() {
         }).verifyComplete();
     }
 
+    @DisabledIf("com.azure.storage.file.share.FileShareTestBase#olderThan20210410ServiceVersion")
+    @Test
+    public void uploadRangeFromURLOAuth() {
+        ShareServiceAsyncClient oAuthServiceClient = getOAuthServiceClientAsyncSharedKey(new ShareServiceClientBuilder()
+            .shareTokenIntent(ShareTokenIntent.BACKUP));
+        ShareDirectoryAsyncClient dirClient = oAuthServiceClient.getShareAsyncClient(shareName)
+            .getDirectoryClient(generatePathName());
+        dirClient.create().block();
+        String fileName = generatePathName();
+        ShareFileAsyncClient fileClient = dirClient.getFileClient(fileName);
+        fileClient.create(1024).block();
+
+        String data = "The quick brown fox jumps over the lazy dog";
+        int sourceOffset = 5;
+        int length = 5;
+        int destinationOffset = 0;
+
+        fileClient.uploadRange(Flux.just(ByteBuffer.wrap(data.getBytes())), data.length()).block();
+        StorageSharedKeyCredential credential = StorageSharedKeyCredential.fromConnectionString(
+            ENVIRONMENT.getPrimaryAccount().getConnectionString());
+        String sasToken = new ShareServiceSasSignatureValues()
+            .setExpiryTime(testResourceNamer.now().plusDays(1))
+            .setPermissions(new ShareFileSasPermission().setReadPermission(true))
+            .setShareName(fileClient.getShareName())
+            .setFilePath(fileClient.getFilePath())
+            .generateSasQueryParameters(credential)
+            .encode();
+
+        String fileNameDest = generatePathName();
+        ShareFileAsyncClient fileClientDest = dirClient.getFileClient(fileNameDest);
+        fileClientDest.create(1024).block();
+
+        StepVerifier.create(fileClientDest.uploadRangeFromUrlWithResponse(length,
```

Comment:
Have we ran these tests with Fiddler enabled to verify this request's x-ms-file-request-intent and x-ms-source-authorization headers are populated?

Reply:
I believe they are being populated. I see "x-ms-file-request-intent: backup" in fiddler. I don't see x-ms-source-authorization, but I do see  "Authorization: SharedKey", if that is what you're referring to.

Reply:
I checked the .NET tests for this, the behavior you are describing is correct.